pkg/reconciler/eventlistener/resources/container.go

```diff
@@ -57,15 +57,19 @@ func MakeContainer(el *v1beta1.EventListener, configAcc reconcilersource.ConfigA
 			Capabilities: &corev1.Capabilities{
 				Drop: []corev1.Capability{"ALL"},
 			},
-			RunAsNonRoot: ptr.Bool(true),
+			RunAsNonRoot: ptr.Bool(cfg.Defaults.DefaultRunAsNonRoot),
 			SeccompProfile: &corev1.SeccompProfile{
 				Type: corev1.SeccompProfileTypeRuntimeDefault,
 			},
 		}
 	}
 
-	containerSecurityContext.RunAsUser = ptr.Int64(cfg.Defaults.DefaultRunAsUser)
-	containerSecurityContext.RunAsGroup = ptr.Int64(cfg.Defaults.DefaultRunAsGroup)
+	if !cfg.Defaults.IsDefaultRunAsUserEmpty {
```

Comment:
can we have this also behind c.SetSecurityContext

Reply:
and other field like capability, priviledge escalation can be moved out of the flag

Reply:
ya make sense

we can do that